### PR TITLE
Revert "Revert "Add support for large integers""

### PR DESCRIPTION
**Pull request by themasch**
Reverts themasch/node-bencode#50

so here we start the new diskussion about #44 after reverting it in #50 ;)

### CODE REVIEW

File: lib/encoder.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+ * @type {Object}
+ */
+Encoder.prototype = {
+
+  constructor: Encoder,
+
+  use: function (type, fn, check) {
+    this[type] = fn
+    this._types.unshift(function (value) {
+      return check(value) ? type : void 0
```
Comment (themasch) [inline lib/encoder.js]: shouldn't we be able to use "undefined" here?

**Comment by themasch** [comment]
I really like the idea of making. I'm currently failing to understand how this could work for decoding but encoding looks pretty good. It also enables lots of other use cases like the binary-key-dict one or real booleans etc. We/people could release bencode-* plugins that extend/replace types in bencode. :+1: from me.

File: lib/encoder.js (hunk 1 above)
Comment (jhermsmeier) [inline lib/encoder.js]: oh my habit of void 0...

**Comment by jimmywarting** [comment]
New ways introduced `1n` and `BigInt(1)`

**Comment by ThaUnknown** [comment]
https://github.com/webtorrent/node-bencode/pull/150

**Pull request closed without merging**